src/model/db.js

```diff
@@ -2,94 +2,82 @@
-					table.increments('ID').primary();
-					table.integer('Owner').references('Users.ID');//.notNullable();  //This shouldn't be nullable, but we don't have users working yet
-					table.integer('GameID').references('Games.ID').nullable();
-					table.string('Name').notNullable();
-					table.boolean('Adult').defaultTo(false);
-					table.string('Description').notNullable().defaultTo('');
-				});
-			}
-			return Promise.resolve();
-		})
-		.then(() => knex.schema.hasTable('ChildBoards'))
-		.then((exists) => {
-			if (!exists) {
-				return knex.schema.createTable('ChildBoards', (table) => {
-					table.increments('ID').primary();
-					table.integer('ParentID').references('Boards.ID').notNullable();
-					table.integer('ChildID').references('Boards.ID').notNullable();
-				});
-			}
-			return Promise.resolve();
-		})
-		.then(() => knex.schema.hasTable('Threads'))
-		.then((exists) => {
-			if (!exists) {
-				return knex.schema.createTable('Threads', (table) => {
-					table.increments('ID').primary();
-					table.string('Title').notNullable();
-					table.integer('Board').references('Boards.ID').notNullable();
-				});
-			}
-			return Promise.resolve();
-		})
-		.then(() => knex.schema.hasTable('Posts'))
-		.then((exists) => {
-			if (!exists) {
-				return knex.schema.createTable('Posts', (table) => {
-					table.increments('ID').primary();
-					table.integer('Thread').references('Threads.ID').notNullable();
-					table.string('Body').notNullable();
-				});
-			}
-			return Promise.resolve();
-		}).then(() => {
-			db.initialised = true;
-			return Promise.resolve(db.initialised);
-		});
+
+		const tables = [
+			['Games', (table) => {
+				table.increments('ID').primary();
+				table.string('gameDescription');
+			}],
+			['Users', (table) => {
+				table.increments('ID').primary();
+				table.string('Username').notNullable().unique();
+			}],
+			['Boards', (table) => {
+				table.increments('ID').primary();
+				//This shouldn't be nullable, but we don't have users working yet
+				table.integer('Owner').references('Users.ID'); //.notNullable();
+				table.integer('GameID').references('Games.ID').nullable();
+				table.string('Name').notNullable();
+				table.boolean('Adult').defaultTo(false);
+				table.string('Description').notNullable().defaultTo('');
+			}],
+			['ChildBoards', (table) => {
+				table.increments('ID').primary();
+				table.integer('ParentID').references('Boards.ID').notNullable();
+				table.integer('ChildID').references('Boards.ID').notNullable();
+			}],
+			['Threads', (table) => {
+				table.increments('ID').primary();
+				table.string('Title').notNullable();
+				table.integer('Board').references('Boards.ID').notNullable();
+			}],
+			['Posts', (table) => {
+				table.increments('ID').primary();
+				table.integer('Thread').references('Threads.ID').notNullable();
+				table.string('Body').notNullable();
+			}]
+		];
+
+		return ensureKnexTables(knex, tables)
```

Comment:
How does this work? i've been scrolling back and forth for a few minutes and I can't place it. It returns a function that would ensure a table exists?